### ISSUE TITLE
Notification can be annoying

### ISSUE THREAD

**Issue by mahotilo** [Typesetter/Typesetter#332]
I find new notification system in TS very useful, but I have some controversial user experience.

I've tried last build with my current project based on 1PTK. So, I need to hide some pages like "Sitemap", "Galleries", "Missing" and "Search" because they break one-page logic and  have no sense in my case. But now I always have quite distracting  numbers near the bell. I think, the Site Explorer addon do this job more unobtrusively, doesn't it? 
Can we show numbers only for those notifications, that need some response from admin (at least for high-level bell)? Maybe we should use an empty coloured circle (or small dot) for notifications like "there are a private pages". On the other hand, I personally prefer to see coloured background for "Visibility: Private" in the top menu to be warned about.

**Comment by juek** [comment]
> Can we show numbers only for those notifications, that need some response from admin

Certainly!
I implement custom filtering methods (per user settings).

I already use a 'Priority' variable to control the display of the number badges. I think I will also use it to hide notes (with negative values) so that plugins can also influence it. Such a "display score" scheme is very flexible - it may e.g. be used for things like automatic increase / decrease of priority.

Notifications are still in progress. For example, The Dashboard widget does not work yet.

**Comment by mahotilo** [comment]
Ok, thanks for clarifying situation. We will wait for the final implementation of the notification system.

**Comment by juek** [comment]
If you don't mind, I'd like to keep the issue open until things are set in stone for a release. I really appreciate feedback for such new features.

**Comment by juek** [comment]
Should we show draft and private page notifications to admins who don't have editing permissions for the particular page(s) or extra area?
I'd say no, because they can't do anything about it?

What do you guys think?

**Comment by mahotilo** [comment]
Hmm, such a limited “vision” can impede to certain types of cooperative work, but this is not a big problem. So yes, I agree. A 'limited' admin does not need to worry about the warnings of a 'super' administrator.

**Comment by juek** [comment]
Latest [commit](https://github.com/Typesetter/Typesetter/commit/079e6460b86193c164923fcd8e8a1bad2bc45dc2) comes with some changes:

* Permission filters: admins without permissions will not see notifications for issues they are not allowed to deal with.
* Custom filters: admins can mark notification items as 'hidden' so they will not cast a number badge nor add to the total number anymore. Filters are stored individually per user, can be reset any time and remain active as long as the notification/item is active.
* The admin box to list/manage notification items now has 'prev' and 'next' navigation to zap through all notifications.

**Comment by juek** [comment]
Screenshot:

![grafik](https://user-images.githubusercontent.com/704283/70098635-e7b51d80-162c-11ea-8a41-90281900b940.png)

**Comment by mahotilo** [comment]
Changing bell state causes screen blinking (the modal form closes and appears again). There is no  smoother way to update state, is it? Or it is work in progress?

**Comment by juek** [comment]
> Changing bell state causes screen blinking 

The whole box will be replaced (together with the Notifications admin panel). 
Maybe I should just implement an additional 'hard-transition' gpabox method for it.

**Comment by mahotilo** [comment]
I suppose it totally acceptable when the number near the "bell" at menu will be changed only after the dialog window is closed. As for me, there is no need for live notification icon update when you click on the visibility selector.

**Comment by juek** [comment]
> I suppose it totally acceptable when the number near the "bell" at menu will be changed only after the dialog window is closed.

Would be more the Typesetter UX we know. The box then would have an [Update] and a [Save] button instead of [Close]. 

Hmm … but on the other side … what happens when we click e.g. a 'View & Edit Page' link?

**Comment by mahotilo** [comment]
> Would be more the Typesetter UX we know.

It sounds great. All I can say: !"May the force be with you"!

> Hmm … but on the other side … what happens when we click e.g. a 'View & Edit Page' link?

target="_blank" can help us:)

**Comment by juek** [comment]
> target="_blank" can help us:)

Nooo! We don't do that. It would leave an open box with unsaved state on the source page - which  would in addition have to auto-update when the notification item was solved (e.g. publishing the draft on the target page).  If it wouldn't auto-update, we might save a display filter for a notification item that wouldn't exist anymore. Too complicated and barely comprehensible behavior IMO.

**Comment by juek** [comment]
I think I'll go for a 'fastbox' AJAX method w/o loader overlay and fadeIn. That's rather easy to implement.

**Comment by mahotilo** [comment]
> It would leave an open box with unsaved state on the source page - which would in addition have to auto-update when the notification item was solved (e.g. publishing the draft on the target page).

But user still can do this by himself. Do we need to block right click (and show our context menu)?

**Comment by juek** [comment]
> Do we need to block right click

No we don't. If a user intentionally wants to keep the page open, that's fine.

**Comment by juek** [comment]
Notification box navigation and filter setting -> flickering fixed with [this commit](https://github.com/Typesetter/Typesetter/commit/31c39ef6f1e7604cd008f4f016787b1552fdf516)

**Comment by mahotilo** [comment]
Yes, I can confirm

**Comment by juek** [comment]
I yet have to test it on a slow/throttled host. We might have to block repeated click attempts if the response is slow.

**Comment by juek** [comment]
added debug notifications with [this commit](https://github.com/Typesetter/Typesetter/commit/dc682ddcf261d3706b834a7c86a541ae7b78c802)

The Notification system will now show some development-relevant information:

* If `ini_set(display_errors,1)` is set in gpconfig.php, this will trigger a high-priority note to all users.
* `define('gpdebug', true)` (rather low priority) will be issued only to 'superusers', namely admins with all permissions granted.
* the latter applies to the brandnew `define('create_css_sourcemaps', true)` (also low priority) 

Of course, all 3 notification item types can be muted, but they will reappear when the site moves to a different host (when $_SERVER['SERVER_NAME'] changes). This will make sure you don't forget to disable debugging measures when going live.

Guys, please tell me if you consider this overkill or if you're missing anything important.

The current priority range is defined as follows:

```
-1 (neg. values) --- 0 --------- 40 -------- 60 ------- 75 ------ 100 ------- 500 
muted notes       none set  private pages  updates  debug notes  drafts  display_errors=1

```

**Comment by juek** [comment]
Screenshot: 
![grafik](https://user-images.githubusercontent.com/704283/70859469-58551780-1f14-11ea-97cf-e7470806721c.png)

**Comment by mahotilo** [comment]
Notification works for me in quite understandable way. 

But...
I don't have much experience in working with gpdebug, so I have played with it. 
I have tried to define gpdebug in this way
```
...
define('gpdebug',false);
define('gpdebug',true);
...
```
and got an empty page after a Ctrl+F5. Is it normal for such "redefining"?

**Comment by juek** [comment]
You cannot redefine a constant. If you could, it would be a variable.

**Comment by mahotilo** [comment]
Oh, thanks. Php can be so scary with its empty silent.
